otter/convergence/model.py

```diff
@@ -126,6 +154,69 @@ def __init__(self):
         assert self.state in ServerState.iterconstants(), \
             "%r is not a ServerState" % (self.state,)
 
+    @classmethod
+    def group_id_from_metadata(cls, metadata):
+        """
+        Get the group ID of a server based on the metadata.
+
+        The old key was ``rax:auto_scaling_group_id``, but the new key is
+        ``rax:autoscale:group:id``.  Try pulling from the new key first, and
+        if it doesn't exist, pull from the old.
+        """
+        return metadata.get(
+            "rax:autoscale:group:id",
+            metadata.get("rax:auto_scaling_group_id", None))
+
+    @classmethod
+    def lbs_from_metadata(cls, metadata):
+        """
+        Get the desired load balancer descriptions based on the metadata.
+
+        :return: ``dict`` of `ILBDescription` providers
+        """
+        lbs = get_service_metadata('autoscale', metadata).get('lb', {})
+        desired_lbs = {}
+
+        for lb_id, v in lbs.get('CloudLoadBalancer', {}).iteritems():
+            # if malformed, skiped the whole key
+            try:
+                configs = json.loads(v)
+                if isinstance(configs, list):
+                    desired_lbs[lb_id] = [
+                        CLBDescription(lb_id=lb_id, port=c['port'])
+                        for c in configs]
+            except (ValueError, KeyError, TypeError):
+                pass
+
+        return pmap(desired_lbs)
+
+    @classmethod
+    def generate_metadata(cls, group_id, lb_descriptions):
+        """
+        Generate autoscale-specific Nova server metadata given the group ID and
+        an iterable of :class:`ILBDescription` providers.
+
+        NOTE: Currently this ignores RCv3 settings and draining timeout
+        settings, since they haven't been implemented yet.
+
+        :return: a metadata `dict` containing the group ID and LB information
+        """
+        metadata = {
+            'rax:auto_scaling_group_id': group_id,
```

Comment:
Yes, it's for backwards compatibility.  Yes, we can only write one in convergence so long as we can get Reach to read both keys for a while (we will have to do the same).

Unless we want to put the effort into also migrating the current worker, and also migrate all existing server keys.